### PR TITLE
Add serde feature

### PR DESCRIPTION
**Pull request by bcmyers**
Would you guys mind accepting this small PR, which adds serde's Serialize and Deserialize traits to AsciiChar and AsciiString behind a serde feature?

### CODE REVIEW

**Comment by tormol** [comment]
Thank you, optional serde support is a no-brainer.  
I don't think adding it is a breaking change either: While serde requires Rust 1.13 and this crate Rust 1.9, adding it as an opt-in feature should not affect current users.
That requirement should be documented in the crate-level documentation (in lib.rs and the readme) though.

On the other hand, *small* is not the word I'd use to describe a "+164 -100" commit.
Could you at least split out the rustfmt run (I assume) into a later commit?
(or drop it completely; I'm not a big fan of rustup in the first place, and most of these changes makes the code longer (=less code on screen) while not improving readability IMO)

**Comment by bcmyers** [comment]
Yes, the other changes are rustfmt, which I'll turn off. I'll also adds some documentation.

**Comment by bcmyers** [comment]
There we go. Three commits:
* One that prevents rustfmt from running (so future contributors with rustfmt turned on in their IDE won't make the same mistake I did)
* Another that adds serialization and deserialization support for `AsciiChar` and `AsciiString` behind a `serde` feature flag, with the minimum Rust version documented.
* Another that bumps the version to 0.9.1 (because, as you mentioned, this is not a breaking change)

File: src/ascii_string.rs [hunk 1]
```diff
@@ -14,6 +16,7 @@ use ascii_str::{AsciiStr, AsAsciiStr, AsAsciiStrError};
 
 /// A growable string stored as an ASCII encoded buffer.
 #[derive(Clone, Default, PartialEq, Eq, PartialOrd, Ord, Hash)]
+#[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
```
Comment (tomprogrammer) [inline src/ascii_string.rs]: Shouldn't this also be annotated with `#[cfg_attr(feature = "serde", serde(transparent))]?

**Comment by tomprogrammer** [comment]
This PR implements serialization exposing the inner structure of the `AsciiString` and `AsciiChar` type.

Example: `{"vec":["t","e","s","t","Space","s","t","r","i","n","g"]}`

I think we should serialize and deserialize to a string (serde datatype).

**Comment by bcmyers** [comment]
Your earlier question prompted me to the same conclusion; so I've been working on a version that uses custom impls of `Serialize` and `Deserialize` to basically have `AsciiString` and `AsciiChar` behave like regular `String` and regular `Char`.

**Comment by bcmyers** [comment]
So here's a version that has a lot more code, but it should make serialization and deserialization work exactly like the standard `Char`, `&'a str`, and `String`types. 

Even though it's relatively complicated code, all I did was basically copy/paste the impls that serde itself has for `Char`, `&'a str`, and `String` and adapt them slightly for the various ascii types; so hopefully it won't be too hard to review.

File: src/serialization/ascii_char.rs [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+
+    #[inline]
+    fn visit_char<E: Error>(self, v: char) -> Result<Self::Value, E> {
+        AsciiChar::from(v).map_err(|_| Error::invalid_value(Unexpected::Char(v), &self))
+    }
+
+    #[inline]
+    fn visit_str<E: Error>(self, v: &str) -> Result<Self::Value, E> {
+        let mut iter = v.chars();
+        match (iter.next(), iter.next()) {
```
Comment (tomprogrammer) [inline src/serialization/ascii_char.rs]: Wouldn't a simple check `v.len() == 1` suffice?

```
if v.len() == 1 {
    self.visit_char(v[0])
} else {
    Err(Error::invalid_value(Unexpected::Str(v), &self))
}
```
This seems simpler to me, although there exists an invariant between the first two lines.

Comment (bcmyers) [reply src/serialization/ascii_char.rs]: Although, we can't get a `char` by indexing a `&str` (i.e. no `v[0]`); so I'll do this instead...

```
        if v.len() == 1 {
            let c = v.chars().next().unwrap();
            self.visit_char(c)
        } else {
            Err(Error::invalid_value(Unexpected::Str(v), &self))
        }
```

Comment (bcmyers) [reply src/serialization/ascii_char.rs]: Unless there's a more elegant way you can think of to get that `char` from the `&str`.

Comment (tomprogrammer) [reply src/serialization/ascii_char.rs]: Ah, you're right.. I should have tested that at the playground.

Interestingly the `char::from_str` function looks just like your code. I'd say, just keep it your way.

**Comment by tomprogrammer** [comment]
I assume that you finished working on this PR? I'd merge it then.

**Comment by bcmyers** [comment]
I don't have any additional changes I'd like to make, but happy to continue incorporating any comments you guys have.

**Pull request merged**